### ISSUE TITLE
com.docker.supervisor failed to start - Exit code 1

### ISSUE THREAD

**Issue by MikeMitterer** [docker/for-mac#2770]
Please, check https://docs.docker.com/docker-for-mac/troubleshoot/.
Issues without logs and details cannot be debugged, and will be closed.

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have submitted Diagnostics
  - [X] I have included the Diagnostics ID in this Issue

### Expected behavior
Docker should start

### Actual behavior
Does not start

### Information
Log comes from: docker-diagnose --last 1d --upload

ID: 5DBB1891-D9D2-4C60-96C0-D2BECBEA8070

[20180404-081659.tar.gz](https://github.com/docker/for-mac/files/1874537/20180404-081659.tar.gz)

The GUI-Diagnose option is not possible because it crashes immediately. 
In the error dialog - if I click on "Diagnose..." it hangs... 

```
[OK]      Files
[ERROR]   Docker CLI       Connection refused (ECONNREFUSED) connecting to /var/run/docker.sock: check if service is running
[ERROR]   Docker CLI       Connection refused (ECONNREFUSED) connecting to /Users/mikemitterer/Library/Containers/com.docker.docker/Data/s60: check if service is running
Error docker ps: timeout after 10.00s
[ERROR]   docker-cli       docker ps failed: (Failure "docker ps: timeout after 10.00s")
[OK]      Moby booted
[ERROR]   driver.amd64-linux com.docker.driver.amd64-linux is not running
[OK]      vmnetd
[ERROR]   osxfs            com.docker.osxfs is not running
[ERROR]   VPNKit           Unexpected error ((Failure
  "Error connecting socket to 9p endpoint unix:/Users/mikemitterer/Library/Containers/com.docker.docker/Data/s51: Unix.Unix_error(Unix.ECONNREFUSED, \"connect\", \"\")")) connecting to /Users/mikemitterer/Library/Containers/com.docker.docker/Data/s51
[ERROR]   VPNKit           com.docker.vpnkit is not running
Error docker ps: got timeout (docker ps: exit 1)
[OK]      disk
[OK]      environment
[ERROR]   Docker           Docker is not running
[OK]      VT-x
[OK]      kern.hv_support
Most specific failure is: Docker is not running
Docker for Mac: version...

```
### Steps to reproduce the behavior

  1. Start Docker

**Comment by MikeMitterer** [comment]
Docker ran smoothly on my machine forever but in the last several month there is on major bug after the other. 

I had to reset Docker to Factory settings - now it works again.
**Annoying**!!!!

**Comment by akimd** [comment]
Hi @MikeMitterer.  Sorry about that.  It's quite tricky to run something so technical on so many different configurations.  For instance, your OS is quite old.

Your logs were not uploaded.  Next time you run diagnose by hand, please let it upload the report, so that our tools can run on them as expected.

Your logs contain
```
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Notice>: Filesharing: /Volumes/DataElCapitan/Typo3/HostEurope/backup:/Volumes/DataElCapitan/Typo3/HostEurope/backup
        /Volumes/Daten/DevLocal/DevSec:/Volumes/Daten/DevLocal/DevSec
        /Volumes/Daten/DevLocal/DevWeb/WordPress/BitCon/docker:/Volumes/Daten/DevLocal/DevWeb/WordPress/BitCon/docker
        /Volumes/Daten/Jenkins:/Volumes/Daten/Jenkins
        /private:/private
        /tmp:/tmp
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Notice>: Done writing to /Applications/Docker.app/Contents/MacOS/com.docker.osxfs
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Notice>: Starting &{/Applications/Docker.app/Contents/Resources/bin/com.docker.vpnkit [com.docker.vpnkit --ethernet fd:3 --port fd:4 --introspection fd:5 --diagnostics fd:6 --vsock-path /Users/mikemitterer/Library/Containers/com.docker.docker/Data/vms/0/connect --host-names host.docker.internal,docker.for.mac.host.internal,docker.for.mac.localhost --gateway-names gateway.docker.internal,docker.for.mac.gateway.internal,docker.for.mac.http.internal --vm-names docker-for-desktop --listen-backlog 32 --mtu 1500 --allowed-bind-addresses 0.0.0.0 --http /Users/mikemitterer/Library/Group Containers/group.com.docker/http_proxy.json --dhcp /Users/mikemitterer/Library/Group Containers/group.com.docker/dhcp.json --port-max-idle-time 300 --max-connections 2000 --gateway-ip 192.168.65.1 --host-ip 192.168.65.2 --lowest-ip 192.168.65.3 --highest-ip 192.168.65.254 --log-destination asl] []  <nil> 0xc420332078 0xc420332088 [0xc4204da4b8 0xc4204da4c0 0xc4204da4c8 0xc4204da4d0] <nil> <nil> <nil> <nil> <nil> false [ [...]ocker.osxfs][2174] <Notice>: Writeback cache is disabled
2018-04-04 08:13:03+02 mac0 Docker[com.docker.osxfs][2174] <Notice>: Cache timeouts set to {"entry_valid_nsec":"10000000","entry_valid":"0","attr_valid_nsec":"0","attr_valid":"0"}
2018-04-04 08:13:03+02 mac0 Docker[com.docker.osxfs][2174] <Error>: Fatal unexpected exception: Unix.Unix_error(Unix.ENOENT, "readlink", "/Volumes/Daten/DevLocal/DevWeb/WordPress/BitCon/docker")
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Notice>: com.docker.osxfs: stderr: osxfs: Unix.Unix_error(Unix.ENOENT, "readlink",
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Notice>: com.docker.osxfs: stderr:        "/Volumes/Daten/DevLocal/DevWeb/WordPress/BitCon/docker")
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Error>: cannot read stdout from com.docker.osxfs: EOF
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Error>: cannot read stderr from com.docker.osxfs: EOF
2018-04-04 08:13:03+02 mac0 Docker[com.docker.supervisor][2173] <Notice>: Reap com.docker.osxfs (pid: 2174): exit status 1
```

Are you sure all the volumes were available when you started Docker?  In particular, `/Volumes/Daten/DevLocal/DevWeb/WordPress/BitCon/docker` is available?  It's definitely sad that this sufficed to kill osxfs, of course.

**Comment by djs55** [comment]
I can reproduce this by
- creating a directory (I used `mkdir /usr/local/bin/broken` since it was writable by default and didn't overlap with an existing shared directory)
- adding it to the volume share list in preferences and restarting
- deleting the directory and replacing it with a dangling symlink (`ln -s /broken broken`)
- restarting the app again

One workaround is to edit `~/Library/Group\ Containers/group.com.docker/settings.json` and remove the path `/Volumes/Daten/DevLocal/DevWeb/WordPress/BitCon/docker`. Another is to make that symlink into a directory (or otherwise make it valid).

I'll investigate making the code robust to this condition in the first place.

**Comment by akimd** [comment]
Fixed in Edge 18.04.
